job.c

```diff
@@ -89,6 +89,27 @@ job_find(uint64 job_id)
     return jh;
 }
 
+job
+job_find_by_body(job j)
```

Comment:
This function is `O(n)` where `n` is number of jobs in queue, right? Unfortunately, that's not acceptable. We want at least `O(log(n))` or better.